reanalysis/utils.py

```diff
@@ -607,51 +606,46 @@ def write_output_json(output_path: str, object_to_write: dict):
         logging.info(f'Output path {output_path!r} exists, will be overwritten')
 
     with out_route.open('w') as fh:
-        json.dump(object_to_write, fh, indent=4, default=str)
+        json.dump(object_to_write, fh, indent=4, default=list)
 
 
-def get_simple_moi(panel_app_moi: str | None) -> str:
+def get_simple_moi(input_moi: str | None, chrom: str) -> str | None:
     """
     takes the vast range of PanelApp MOIs, and reduces to a
     range of cases which can be easily implemented in RD analysis
-    This is required to reduce the complexity of an MVP
-    Could become a strict enumeration
 
-    {
-        Biallelic: [all, panelapp, moi, equal, to, biallelic],
-        Monoallelic: [all MOI to be interpreted as monoallelic]
-    }
+    Args:
+        input_moi ():
+        chrom ():
+
+    Returns:
 
-    :param panel_app_moi: full PanelApp string or None
-    :return: a simplified representation
     """
 
-    # default to considering both. NOTE! Many genes have Unknown MOI!
-    simple_moi = 'Mono_And_Biallelic'
-
-    # try-except permits the moi to be None
-    try:
-        lower_moi = panel_app_moi.lower()
-        if lower_moi is None or lower_moi == 'unknown':
-            # exit iteration, all moi considered
-            return simple_moi
-    except AttributeError:
-        return simple_moi
-
-    # ideal for match-case, coming to a python 3.10 near you!
-    if lower_moi.startswith('biallelic'):
-        simple_moi = 'Biallelic'
-    elif lower_moi.startswith('both'):
-        simple_moi = 'Mono_And_Biallelic'
-    elif lower_moi.startswith('mono'):
-        simple_moi = 'Monoallelic'
-    elif lower_moi.startswith('x-linked'):
-        if 'biallelic' in panel_app_moi:
-            simple_moi = 'Hemi_Bi_In_Female'
-        else:
-            simple_moi = 'Hemi_Mono_In_Female'
-
-    return simple_moi
+    if input_moi in IRRELEVANT_MOI:
+        raise ValueError("unknown and other shouldn't reach this method")
+
+    default = 'Hemi_Mono_In_Female' if chrom in X_CHROMOSOME else 'Mono_And_Biallelic'
```

Comment:
Am I correct that this will be applied if there is no proper MOI in any of the panels used? If so, I am worried that this will end up being too verbose. It may be better to spit our a "no MOI" warning then default to:
```Hemi_Bi_In_Female' if chrom in X_CHROMOSOME else 'Biallelic'```

Reply:
Updated this PR to use biallelic/Hemi and Biallelic as the base MOI. The default could also be taken from the config, but I don't think we'll be coming across this situation that frequently.